include/class.mailer.php

```diff
@@ -66,10 +66,12 @@ function setFromAddress($from) {
         $this->ht['from'] = $from;
     }
 
-    function getFromAddress() {
+    function getFromAddress($options=array()) {
 
-        if(!$this->ht['from'] && ($email=$this->getEmail()))
-            $this->ht['from'] =sprintf('"%s" <%s>', ($email->getName()?$email->getName():$email->getEmail()), $email->getEmail());
+        if (!$this->ht['from'] && ($email=$this->getEmail())) {
+            $name = $options['from_name'] ?: ($email->getName() ?: $email->getEmail());
+            $this->ht['from'] =sprintf('"%s" <%s>', $name, $email->getEmail());
```

Comment:
I don't think `$name` here should fallback to the email address, although I see it was previously written that way. Perhaps
```php
    if ($name = $options['from_name'] ?: $email->getName())
        $this->ht['from'] = sprintf('"%s" <%s>', $name, $email->getEmail());
    else
        sprintf('<%s>', $email->getEmail());